host/windows/hostthread.c

```diff
@@ -13,6 +13,22 @@
 **==============================================================================
 */
 
+int oe_thread_create(oe_thread* thread, void* (*func)(void*), void* arg)
+{
+    *thread = CreateThread(NULL, 0, func, arg, 0, NULL);
+    return *thread == NULL;
+}
+
+int oe_thread_join(oe_thread thread)
+{
+    if (WaitForSingleObject(thread, INFINITE) == WAIT_OBJECT_0)
+    {
+        CloseHandle(thread);
+        return 0;
+    }
+    return 1;
```

Comment:
Like Mike said in another thread (no pun intended), the error codes returned by the Windows implementation and Linux implementation are so different thus we can't make a cross-platform interpretation of them. However it may still be desirable to let the caller know whether this function succeeds or not.

I can make it return either `OE_OK` or `OE_THREAD_JOIN_ERROR`, does that address your concerns?

Reply:
I've said this elsewhere but I'll repeat it again here.   I do not think we should have an oe_thread_join per se.   We should instead have oe_pthread_join that exposes the POSIX syntax (including on Windows) just prefixed with oe.  In that implementation it would return the POSIX error codes not oe_result_t error codes.

Reply:
`oe_thread_create` and `oe_thread_join` return POSIX error codes on Linux now. On Windows they are mapped to the POSIX error codes as much as possible.